interface/karma/karma-unit.tpl.js

```diff
@@ -35,6 +35,8 @@ module.exports = function ( karma ) {
      * How to report, by default.
      */
     reporters: 'dots',
+    // This option is handy for sorting out some kinds of testing errors
+    // logLevel: karma.LOG_DEBUG,
```

Comment:
Karma like to be terse as it runs through the tests, most likely for efficiency. If the tests aren't working right, though, it can be maddening! This setting makes karma itself tell you more about what it's doing.

To get more debug info out of the tests themselves, I was able to use `console.log()` statements to good effect, along with explicitly reporting what the mock `$log` service captured as I've done in the `.spec.js` files.